File: Problem4/Problem4.py
# created by adanai at 2019-12-02 10:59.
# 
# Given an array of integers, return a new array such that,
# each element at index i of the new array is the product of 
# all the numbers in the original array except the one at i

arr = [1, 2, 3, 4, 5] #defining a random array

def arr_prod(l): #function definition for the problem
    prod = 1 #initializing a product variable
    for i in l:
        prod = prod*i #obtaining the product of the array
    #creating an array of the size of given array 
    # with identical elements equal to the product
    fin_arr = [prod]*len(l) 
    #dividing the product in fin_arr by element of arr
    fin_arr = [fin_arr[i]//l[i] for i in range(0, len(l))]
    return fin_arr
    
print(arr_prod(arr))

File: Problem4/Problem4.java
/*
 Given an array of integers, return a new array such that,
 each element at index i of the new array is the product of 
 all the numbers in the original array except the one at i
 Note :- Don't use division..	
*/
import java.util.*;
class Problem4 {
	public static void main(String[] args) {
		// Scanner kb = new Scanner(System.in);
		int []arr = new int[]{1, 2, 3, 4, 5};
		// 1 2 3 4 5 
		// 1
		int []ans = new int[5];
		for (int i = 0; i < arr.length; i++) {
			int prod = 1;
			for (int j = 0; j < arr.length; j++) {
				if (i!=j)
					prod = prod * arr[j];
			}
			ans[i] = prod;
		}
		for (int i : ans) System.out.print(i + " ");
 	}
}